provisioning/device/devdoc/transport_state_machine_requirements.md

```diff
@@ -0,0 +1,81 @@
+# transport_state_machine Requirements
+
+## Overview
+This module provides HTTP protocol support to communicate with the Azure device provisioning service
+
+## Example Usage
+``js
+  // this is effectively an abstract class that transport implements are built upon.  It is not meant to be used directly.
+``
+
+## Public Interface
+
+### constructor()
+The constructor creates a TransportStateMachine object
+
+**SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_001: [** The `constructor` shall accept no arguments **]**
+
+
+### connect(callback: (err?: Error) => void): void
+Connect is used to connect the transport
+
+**SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_002: [** `connect` shall call `_doConnectForFsm`. **]**
+
+**SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_003: [** If `_doConnectForFsm` fails, then `connect` shall fail. **]**
```

Comment:
those look like internal implementation details - unnecessary

Reply:
Are you talking about references to all the _doBlahForFsm functions?  In that case, I disagree because that's part of the contract between this class and the transports that extend this class.

Or, are you talking specifically about the "if blah fails, then baz shall fail" part?  In that case, I can probably convince myself that you're right.

---
In reply to: [146396235](https://github.com/Azure/azure-iot-sdk-node/pull/115#discussion_r146396235) [](ancestors = 146396235)